doc/design/design_model_zoo_pai.md

```diff
@@ -0,0 +1,97 @@
+In SQLFlow, we train a model with the following SQLFlow extened SELECT statement:
+```sql
+SELECT * FROM train_table ... TO TRAIN MyAwesomeEstimator ... INTO my_awesome_model;
+```
+and predict with:
+```sql
+SELECT * FROM pred_table ... TO PREDICT using my_awesome_model ... ;
+```
+and analyze with:
+```sql
+SELECT * FROM train_table ...  ANALYZE my_awesome_model USING TreeExplainer;
+
+```
+The extended SELECT statements are executed as described in [Model Zoo Design Doc](./design_model_zoo.md#Submitter-Programs):
+- If the model **is not** from a model zoo, The SQLFlow server converts these statements to a *submitter* file and submits the file to a specified engine.
+- If the model **is** from a model zoo, the submmiter program is mapped into a Docker image. the SQLFlow server calls k8s API to launch the Docker image on a k8s cluster.
+
+## The Design
+### Base Image
+The base image of the SQLFlow model zoo should incorporate both odpscmd(which is already in place) and pyodps.  We propose to use `odpscmd` at the moment because it requires less code modification in both python and golang.
+
+### Submit a SQLFlow Task to PAI With Model Zoo
+When a user submits a SELECT statement as above, SQLFlow should take the following actions:
+1. SQLFlow Checks whether the entity after `TO TRAIN/PREDICT/ANALYZE` is from a SQLFlow model zoo. For example, a plain `DNNClassier` implies that the model is a premade estimator, and `"models.sqlflow.org/sqlflow/my_awesome_model"` implies that the model is from model zoo "models.sqlflow.org". The actual mechanism may be more complicated and is still under progress.
+1. Case A, the model **is not** from a model zoo:
+    1. Case A.1: the user specifies that the task should be submitted to PAI (For example, the user set a environment variable such as `SQLFLOW_submitter=PAI`).
+        - The SQLFlow server generates a submitter script with PAI-required command line options.
+		- The SQLFlow server uses odpscmd or pyodps to submit the script to PAI as [described above](#Submit-a-Task-to-PAI). 
+    1. Case A.2: the task is not to be submitted to PAI:
+	    - Run the other way.
+1. Case B: the model **is** from a model zoo:
+    1. Case B.1: the user specifies that the task should be submitted to PAI.
+    1. Case B.2: the user doesn't specify where to submit but the model knows where. For example, a model zoo table field may specify the platform to submit to.
+	    - The SQLFlow server generates a submitter script with PAI-required command line options.
+        - SQLFlow pulls the Docker image and calls k8s API to launch it on a k8s cluster to execute the following command:
+        ```bash
+        docker run --rm -it \
+          -v /var/sqlflow/submitters:/submitters sqlflow/my_awesome_model \
+            odpscmd -Dscript=/submitters/sqlflow/my_awesome_model.py ...
+        ```
+    1. Case B.3: the task is not to be submitted to PAI.
+	    - Run the other way.
+
+### PAI Model Zoo Data Schema
```

Comment:
Is this the format of the saved model? How this format of the saved model can be published to the model zoo, and how can a user submit a PAI training job using a published model from `model.sqlflow.org` which is of different format?

Reply:
This is only for the PAI model zoo. The first several fields are consistent with models from model.sqlflow.org, so we can simply omit the extra fields 
 when publishing a model to `model.sqlflow.org`, because they are only useful for users of PAI model zoo. Similarly, we can keep the extra fields unfilled when submitting a PAI job using a published model from `model.sqlflow.org`.